File: 2580-circular-sentence/2580-circular-sentence.py
class Solution:
    def isCircularSentence(self, s: str) -> bool:
        n = len(s) 
        if(s[0]!=s[n-1]):
            return False
        prev =" "
        for i in range(n-1):
            if(s[i]==" "):
                if(s[i+1]!=prev):
                    return False 
            prev = s[i]
        return True 

File: 2580-circular-sentence/2580-circular-sentence.java
class Solution {
    public boolean isCircularSentence(String s) {
        if(s.charAt(0)!=s.charAt(s.length()-1)){
            return false ;
        }
        char prev = 0 ; 
        for(int i = 0 ; i<s.length() -1 ; i++){
            if(s.charAt(i)==' '){
                if(s.charAt(i+1)!=prev){
                    return false ;
                }
            }
            prev = s.charAt(i) ;
        }
        return true ;
    }
}